stats/stats/src/charts/counters/completed_txns.rs

```diff
@@ -13,31 +16,57 @@ use sea_orm::{DbBackend, Statement};
 pub struct CompletedTxnsStatement;
 
 impl StatementForOne for CompletedTxnsStatement {
-    fn get_statement() -> Statement {
-        Statement::from_string(
-            DbBackend::Postgres,
-            r#"
-                SELECT
-                    (all_success - all_success_dropped)::TEXT AS value,
-                    last_block_date AS date 
-                FROM (
-                    SELECT (
-                        SELECT COUNT(*) AS all_success
-                        FROM transactions t
-                        WHERE t.status = 1
-                    ), (
-                        SELECT COUNT(*) as all_success_dropped
-                        FROM transactions t
-                        JOIN blocks b ON t.block_hash = b.hash
-                        WHERE t.status = 1 AND b.consensus = false
-                    ), (
-                        SELECT MAX(b.timestamp)::DATE AS last_block_date
-                        FROM blocks b
-                        WHERE b.consensus = true
-                    )
-                ) AS sub
-            "#,
-        )
+    fn get_statement(completed_migrations: &BlockscoutMigrations) -> Statement {
+        if completed_migrations.denormalization {
+            Statement::from_string(
+                DbBackend::Postgres,
+                r#"
+                    SELECT
+                        (all_success - all_success_dropped)::TEXT AS value,
+                        last_block_date AS date 
+                    FROM (
+                        SELECT (
+                            SELECT COUNT(*) AS all_success
+                            FROM transactions t
+                            WHERE t.status = 1
+                        ), (
+                            SELECT COUNT(*) as all_success_dropped
+                            FROM transactions t
+                            WHERE t.status = 1 AND t.block_consensus = false
+                        ), (
+                            SELECT MAX(b.timestamp)::DATE AS last_block_date
+                            FROM blocks b
+                            WHERE b.consensus = true
+                        )
```

Comment:
So, the query in this case would be
```sql
SELECT
    (all_success_not_dropped)::TEXT AS value,
    last_block_date AS date 
FROM (
    SELECT (
        SELECT COUNT(*) as all_success_not_dropped
        FROM transactions t
        WHERE t.status = 1 AND t.block_consensus = true
    ), (
        SELECT MAX(b.timestamp)::DATE AS last_block_date
        FROM blocks b
        WHERE b.consensus = true
    )
) AS sub
```

I ran explain for current and suggested queries. Here are the results:
Current:
`Result  (cost=12420.87..12420.89 rows=1 width=36)`
Suggested:
`Result  (cost=23127.56..23127.58 rows=1 width=36)`

According to the rest of the input, the most expensive part of the current query is
```sql
SELECT COUNT(*) AS all_success
    FROM transactions t
    WHERE t.status = 1
```
which is scanned using index: `Parallel Index Only Scan using transactions_status_index on transactions t  (cost=0.42..11158.67 rows=102828 width=0)`. The other part is run very quickly because the `transactions_block_consensus_index` is utilized, which yields negligible amount of transactions (with consensus `false`).

In the suggested query, the expensive part is done twice as costly: `Parallel Seq Scan on transactions t  (cost=0.00..21869.82 rows=102828 width=0)`, likely because the index can't be utilized (for filter `Filter: (block_consensus AND (status = 1))`).

Reply:
After repeated `ANALYZE` runs the difference is even more apparent: in current query time lies in range~ 20..30, whereas for the suggested it's 10 times larger (200..300).
So, the current version better utilizes indexes that are present in `transactions` table. Apparently, they were created similarly to `blocks` table when denormalization of the table took place.